File: sample-github/src/main/java/com/androidhuman/example/samplegithub/ui/main/MainActivity.java
package com.androidhuman.example.samplegithub.ui.main;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.androidhuman.example.samplegithub.R;
import com.androidhuman.example.samplegithub.ui.search.SearchActivity;

public class MainActivity extends AppCompatActivity {

    FloatingActionButton btnSearch;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnSearch = findViewById(R.id.btnActivityMainSearch);

        // 버튼 setOnClickListener 등록
        btnSearch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, SearchActivity.class));
            }
        });
    }
}

File: sample-github-mvp/src/main/java/com/example/sample_github_mvp/view/main/MainActivity.kt
package com.example.sample_github_mvp.view.main

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.sample_github_mvp.R
import com.example.sample_github_mvp.view.search.SearchActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // java와 달리, 코틀린 안드로이드 익스텐션으로 findViewById 사용하지 않고 바로 맵핑 가능
        btnActivityMainSearch.setOnClickListener { startActivity(Intent(this@MainActivity, SearchActivity::class.java)) }
    }
}
